src/main/java/updater/DownloadFile.java

```diff
@@ -0,0 +1,104 @@
+package updater;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URLConnection;
+import java.util.function.Consumer;
+
+/**
+ * A class in charge of downloading file.
+ */
+public class DownloadFile {
+    private static final Logger logger = LogManager.getLogger(DownloadFile.class.getName());
+
+    public static final int CONNECTION_TIMEOUT = 15000;
+    public static final int READ_CONNECTION_TIMEOUT = 30000;
+
+    private final URI source;
+    private final File dest;
+    private long totalFileSizeInBytes;
+
+    private final Consumer<Double> onDownload;
+
+    /**
+     * @param source URI of file source, expect to be a valid URL
+     * @param dest local destination file
+     * @param onDownload callback that listen to download stream
+     */
+    public DownloadFile(URI source, File dest, Consumer<Double> onDownload) {
+        this.source = source;
+        this.dest = dest;
+        this.onDownload = onDownload;
+    }
+
+    /**
+     * Start download from source to destination as specified in constructor
+     *
+     * @return true if download is successful, else false
+     */
+    public boolean startDownload() {
```

Comment:
Yeah. Would renaming it to `download` be better? I will change the private function that actually runs the download to something else.